### ISSUE TITLE
Feature request: improve web interface search ability and other minor issues.

### ISSUE THREAD

**Issue by viperfx** [distribworks/dkron#535]
First of all, thanks for making this. After wrestling with Celery and libraries like Huey for cron related jobs, I am turning to this library for more stability. I have been running it in parallel with the schedulers to understand the stability and reliability. The results so far look great and looking forward to rolling it out to production in future. I plan to purchase the PRO plan too, even though I don't really need most of its features. Just want to support development.

**Ideas**
------

The dashboard displays a "Failed executions" stat. It would be great to click on this, and for the job list to be filtered and show the jobs that had failed executions. 

Why? When you have over 600 jobs, it becomes hard to find the ones you want.

------

Offer a metadata field on the job, so we can store an object of key/value pairs, that might be useful for tracking what that job belongs to. Especially when you need to a trace a job, across other database IDs or other table IDs. 

------

When adding new jobs, I find that the web interface shows that exact amount in "Failed Jobs". As the jobs start to run, they "Failed Jobs" start to go down and the "Successful jobs" start to go up.

Why is this the case? I think "Failed Jobs" should be 0 if no job executions have not been run yet.

**Comment by victorcoder** [comment]
Thanks for your kind comments @viperfx. 


I like the idea of filtering jobs by status, thanks for the suggestion.
---
Interesting, could you elaborate more on the metadata field usage?
---
I should look into this.

**Comment by viperfx** [comment]
@victorcoder I can discuss more details and sensitive data via email. Do you have a preferred email?

**Comment by victorcoder** [comment]
Sure send me an email to victor at victorcastell.com

**Comment by viperfx** [comment]
@victorcoder I sent you an email.

As a more urgent issue 

![image](https://user-images.githubusercontent.com/328257/59456547-8f6f3280-8de4-11e9-91b3-67d8b8d46152.png)

Do you know why it says failed jobs? I just added all these jobs through the API.

**Comment by fopina** [comment]
@victorcoder 
I also have a usecase for metadata fields: in my application I have "temporary" jobs (dkron is simply used to run an async task, not recurrent one) and actual recurrent ones.
One of the recurrent ones, deletes "temporary" jobs that already executed more than 7 days ago to keep list clean.
To mark jobs as those "temporary", I could have gone with name prefix but instead went with a tag but had to add the tag to all agents/workers as well. If there was a metadata field, it'd be cleaner, serving precisely situations like this

@viperfx all jobs without any execution are considered "failed" in the dashboard, not a big issue as they start running the stats become accurate :)

**Comment by victorcoder** [comment]
@viperfx Could you show a sample json of the jobs you are setting? Newly created jobs should not show as failed.

**Comment by victorcoder** [comment]
@fopina we're mixing issues here, there is a good duplicate here https://github.com/victorcoder/dkron/issues/542

I like the idea of a new metadata set to filter jobs, probably the way to go.

**Comment by viperfx** [comment]
```
{
  "name": "member_task_0rxfov_5061",
  "timezone": "America/Toronto",
  "schedule": "0 00 10 * * 1,2,3,4,5",
  "owner": "Mohit Manjaria",
  "owner_email": "",
  "success_count": 0,
  "error_count": 0,
  "last_success": "0001-01-01T00:00:00Z",
  "last_error": "0001-01-01T00:00:00Z",
  "disabled": true,
  "tags": null,
  "retries": 0,
  "dependent_jobs": null,
  "parent_job": "",
  "processors": {
    "log": {
      "forward": true
    }
  },
  "concurrency": "allow",
  "executor": "http",
  "executor_config": {
    "body": "*****",
    "debug": "true",
    "expectCode": "200",
    "headers": "[\"Content-Type: application/json\"]",
    "method": "POST",
    "timeout": "30",
    "url": "*****"
  },
  "status": "success",
  "next": "2019-06-17T10:00:00Z"
}
```

@victorcoder here is an example, from the dkron dashboard.

**Comment by victorcoder** [comment]
Thanks @viperfx, but I don't mean how do you see it in the UI, but what are you sending to the API, because you can insert jobs with `"error_count": 10` and it will count as errors

**Comment by viperfx** [comment]
OK, here is my python code for sending to the API.

```
        exec_body = {
            "args": [report_obj.id],
            "team_slug": report_obj.team.slug,
            "report_slug": report_obj.report_id,
            "schedule": cron_exp,
            "task_type": "report",
            "timezone": str(report_obj.data["config"]["time_zone"]),
        }
        payload = {
            "name": f"report_task_{report_obj.report_id.lower()}",
            "schedule": cron_exp,
            "timezone": str(report_obj.data["config"]["time_zone"]),
            "owner": report_obj.team.user.name,
            "disabled": not report_obj.task.enabled,
            "concurrency": "allow",
            "executor": "http",
            "executor_config": {
                "method": "POST",
                "url": f"{settings.SITE_URL}/webhook/dkron_events/",
                "headers": '["Content-Type: application/json"]',
                "body": json.dumps(exec_body),
                "timeout": "30",
                "expectCode": "200",
                "debug": "true",
            },
            "processors": {"log": {"forward": True}},
        }
        self._consumer.send(
            "POST",
            f"{self.base_url}/jobs/",
            json.dumps(payload, cls=self._serializer),
            self.headers,
        )
```

```
        exec_body = {
            "args": [report_obj.id, member.get("id"), member.get("name")],
            "member_email": member.get("email"),
            "task_type": "member",
            "team_slug": report_obj.team.slug,
            "report_slug": report_obj.report_id,
            "schedule": cron_exp,
            "timezone": str(member_periodic_task.crontab.timezone),
        }
        payload = {
            "name": f"member_task_{report_obj.report_id.lower()}_{member_periodic_task.id}",
            "schedule": cron_exp,
            "timezone": str(timezone),
            "owner": member.get("name"),
            "disabled": not member_periodic_task.enabled,
            "concurrency": "allow",
            "executor": "http",
            "executor_config": {
                "method": "POST",
                "url": f"{settings.SITE_URL}/webhook/dkron_events/",
                "headers": '["Content-Type: application/json"]',
                "body": json.dumps(exec_body),
                "timeout": "30",
                "expectCode": "200",
                "debug": "true",
            },
            "processors": {"log": {"forward": True}},
        }
        self._consumer.send(
            "POST",
            f"{self.base_url}/jobs/",
            json.dumps(payload, cls=self._serializer),
            self.headers,
        )
```

@victorcoder

**Comment by viperfx** [comment]
hey @victorcoder do you any issues with my API request that could be causing the issues?

**Comment by yvanoers** [comment]
@viperfx there is nothing wrong with your API request w.r.t. new jobs counting as failed in the dashboard. New jobs have their last_success and last_error times exactly the same, but only those with last_success > last_error count as successful, everything else counts as failed.
A fix for this is in PR #555.

**Comment by viperfx** [comment]
Thanks for the update @yvanoers 

Look forward to getting it merged soon.

**Comment by victorcoder** [comment]
Closing this because of the issues been resolved, a new issue for the feature request could be opened @viperfx
